src/Service/Modules/AuthenticationModule.cs

```diff
@@ -0,0 +1,71 @@
+﻿namespace Linn.Tax.Service.Modules
+{
+    using Linn.Common.Configuration;
+    using Linn.Tax.Proxy;
+    using Linn.Tax.Resources;
+
+    using Nancy;
+    using Nancy.ModelBinding;
+    using Nancy.Responses;
+
+    public sealed class AuthenticationModule : NancyModule
+    {
+        private readonly IHmrcApiService apiService;
+
+        public AuthenticationModule(IHmrcApiService apiService)
+        {
+            this.apiService = apiService;
+            this.Get("/auth", _ => this.AuthRedirect());
+            this.Get("/tax/signin-oidc-silent", _ => this.SilentRenew());
+            this.Get("/tax/redirect", _ => this.AuthReturn());
+            this.Get("/tax/return", _ => this.SubmitTaxReturn());
+        }
+
+        private object SilentRenew()
+        {
+            return this.Negotiate.WithView("silent-renew");
+        }
+
+        private object AuthRedirect()
+        {
+            var root = ConfigurationManager.Configuration["HMRC_API_ROOT"];
+            var clientId = ConfigurationManager.Configuration["CLIENT_ID"];
+            var redirect = "http://localhost:61798/tax/redirect";
+            var location = $"{root}/oauth/authorize?response_type=code&client_id={clientId}&redirect_uri={redirect}&scope=write:vat";
+            return new RedirectResponse(location);
```

Comment:
Hardcoded scope here... If we only ever use this for one thing (submitting tax returns) then this could be ok.